assemblers/org.osbuild.oci-archive

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/python3
+
+import datetime
+import json
+import os
+import subprocess
+import sys
+import tempfile
+
+
+DEFAULT_PATH = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+
+STAGE_DESC = "Assembler an OCI image archive"
+STAGE_INFO = """
+Assemble an Open Container Initiative[1] image[2] archive, i.e. a
+tarball whose contents is in the OCI image layout.
+
+Currently the only required options are `filename` and `architecture`.
+The execution parameters for the image, which then should form the base
+for the container, can be given via `config`. They have the same format
+as the `config` option for the "OCI Image Configuration" (see [2]),
+except the those that are map to the "Go type map[string]struct{}",
+which are represented as array of strings.
+
+The final resulting tarball, aka a "orci-archive", can be imported via
+podman[3] with `podman pull oci-archive:<archive>`.
+
+[1] https://www.opencontainers.org/
+[2] https://github.com/opencontainers/image-spec/
+[3] https://podman.io/
+"""
+STAGE_OPTS = """
+"additionalProperties": false,
+"required": ["architecture", "filename"],
+"properties": {
+  "architecture": {
+    "description": "The CPU architecture of the image",
+    "type": "string"
+  },
+  "filename": {
+    "description": "Resulting image filename",
+    "type": "string"
+  },
+  "config": {
+    "description": "The execution parameters",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "Cmd": {
+        "type": "array",
+        "default": ["sh"],
+        "items": {
+          "type": "string"
+        }
+      },
+      "Env": {
+        "type": "array",
+        "default": ["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],
+        "items": {
+          "type": "string"
+        }
+      },
+      "ExposedPorts": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "User": {
+        "type": "string"
+      },
+      "Labels": {
+        "type": "object",
+        "additionalProperties": true
+      },
+      "StopSiganl": {
+        "type": "string"
+      },
+      "Volumes": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "WorkingDir": {
+        "type": "string"
+      }
+    }
+  }
+}
+"""
+
+
+MEDIA_TYPES = {
+    "layer": "application/vnd.oci.image.layer.v1.tar",
+    "manifest": "application/vnd.oci.image.manifest.v1+json",
+    "config": "application/vnd.oci.image.config.v1+json"
+}
+
+
+def sha256sum(path: str) -> str:
+    ret = subprocess.run(["sha256sum", path],
+                         capture_output=True,
+                         encoding="utf-8",
+                         check=True)
```

Comment:
`subprocess.check_output()`?

Alternatively: `hashlib`?

Reply:
I think `hashlib` in a future iteration is a good idea, after we add some generic utility class, because we do the same in `sources/org.osbuild.files`.